status-im/status-react
# Reject typeddata request when wrong chainId inside typed data

fixes #20757

### Summary

We should reject any request where the typed data includes the wrong chainId (to prevent replay attacks). As a measure of security, we are checking that the typed data has the correct chain-id and reject potentially malicious requests without showing them to the user. Instead, we just show a toast, informing them that the signing is intended for a different network.

Included in the PR:
* `:wallet-connect/send-response` no longer dismissed the modal, not it's done in `:wallet-connect/finish-session-request`. This is done because sometimes we need to reject (with send-response) without having a modal open, which breaks navigation if we dismiss a modal that doesn't exist.
* rejecting requests when processing a request fails
* rejecting a request when signing/transaction fails
* checking the chainId inside the typeddata to make sure it's not different from the chainId of the session. If different, we show a toast informing the user about it.

### Steps to test


1. Make sure you're on the testnet
2. Connect on https://react-app.walletconnect.com/
3. Send a `eth_signTypedData_v4`
4. A toast is shown and the request is rejected



status: ready
